resources/tentative/complex-static-html/src/react-todomvc/todo/components/item.jsx

```diff
@@ -0,0 +1,51 @@
+import { memo, useState, useCallback } from "react";
+import classnames from "classnames";
+
+import { Input } from "./input";
+
+export const Item = memo(function Item({ todo, dispatch, index }) {
+    const [isWritable, setIsWritable] = useState(false);
+    const { title, completed, id } = todo;
+
+    const toggleItem = useCallback(() => dispatch({ type: "TOGGLE_ITEM", payload: { id } }), [dispatch]);
+    const removeItem = useCallback(() => dispatch({ type: "REMOVE_ITEM", payload: { id } }), [dispatch]);
+    const updateItem = useCallback((id, title) => dispatch({ type: "UPDATE_ITEM", payload: { id, title } }), [dispatch]);
+
+    const handleDoubleClick = useCallback(() => {
+        setIsWritable(true);
+    }, []);
+
+    const handleBlur = useCallback(() => {
+        setIsWritable(false);
+    }, []);
+
+    const handleUpdate = useCallback(
+        (title) => {
+            if (title.length === 0)
+                removeItem(id);
+            else
+                updateItem(id, title);
+
+            setIsWritable(false);
+        },
+        [id, removeItem, updateItem]
+    );
+
+    return (
+        <li className={classnames({ completed: todo.completed }) + ` li-${index}-0`} data-testid="todo-item">
```

Comment:
nit: could we make this more "react'isch"? 

`<li className={classnames({ completed: todo.completed }, `li-${index}-0`)} data-testid="todo-item">`

Reply:
Done